libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyManager.java

```diff
@@ -485,18 +496,24 @@ Optional<Class<?>> findRequestingClass(Stream<Class<?>> classes) {
      * @return true if permission is granted regardless of the entitlement
      */
     private static boolean isTriviallyAllowed(Class<?> requestingClass) {
-        if (logger.isTraceEnabled()) {
-            logger.trace("Stack trace for upcoming trivially-allowed check", new Exception());
-        }
         if (requestingClass == null) {
-            logger.debug("Entitlement trivially allowed: no caller frames outside the entitlement library");
+            // This is unexpected and unusual; we probably want to log at the info level
+            logger.info("Entitlement trivially allowed: no caller frames outside the entitlement library", new Exception("STACK TRACE"));
             return true;
         }
         if (systemModules.contains(requestingClass.getModule())) {
-            logger.debug("Entitlement trivially allowed from system module [{}]", requestingClass.getModule().getName());
+            if (logger.isTraceEnabled()) {
+                logger.trace(
+                    "Entitlement trivially allowed from system module [{}]",
+                    requestingClass.getModule().getName(),
+                    new Exception("STACK TRACE")
```

Comment:
If we want a stack trace, we could generate and print it ourselves? This seems like an abuse of exceptions. But similar to the above comment, it seems unrelated to thread entitlements, can you split these logging changes out?

Reply:
You want a separate PR just to tweak logging?

Reply:
Yes, it's completely unrelated. Let's be able to have an independent discussion on the best way to achieve the logging goals (eg without abusing exceptions to get a stack trace).

Reply:
The "abuse" is already in the code, I just moved it to somewhere more convenient, so it appears in the same log message describing what happened.

I don't understand the alternative you're proposing. How would one generate a stack trace and pass it to the logger?

Reply:
Seeing an exception called "STACK_TRACE" is confusing, and is an abuse because there isn't really an exception, you just want to log a stack trace. You can get the current stack with `Thread.currentThread().getStackTrace()`. But let's debate these two approaches on a separate PR.